c/src/metta.rs

```diff
@@ -1412,6 +1412,24 @@ pub extern "C" fn grounded_number_to_longlong(n: *const atom_t, res: *mut c_long
     }
 }
 
+#[no_mangle]
+pub extern "C" fn grounded_bool_to_int(n: *const atom_t, res: *mut c_int) -> bool {
```

Comment:
`to_int` is no longer accurate in light of above change/ suggestion, changing to `to_bool`

Across the rest of the API, `into` implies consuming the source and converting, where `get` implies access.  `to` is ambiguous and could be read either way.  My personal preference is for `grounded_bool_get_bool`, but it's just a style thing.

Reply:
On C API side it is possible to implement both consuming or getting API. I was going to implement consuming API, thus I would use signature `grounded_bool_into_bool(b: *mut atom_t, res: *mut bool) -> bool {`.

Unfortunately we cannot consume on a Python side. Thus the only viable option to use consuming API in Python is cloning atom before calling a function.

Reply:
> grounded_bool_into_bool(b: *mut atom_t, res: *mut bool) -> bool

The convention we have for consuming is to pass by value.  ie. `grounded_bool_into_bool(b: atom_t, res: *mut bool) -> bool`

However, I don't see the benefit of a consuming converter over a non-consuming accessor, followed by an explicit `free` if the user is done with the atom.  Efficiency-wise they're basically identical.

It's really a style thing, but my philosophy with the C api is to keep the number of functions that consume their arguments to a minimum - if all else is equal.

The other reason for a non-consuming accessor is that it allows you to pass an `atom_ref_t`, like you might get from `atom_vec_get`, the space_t iterator's next function, etc.

Reply:
The reason I would prefer consuming accessor is that if `into` function need to return a more complex object then getter may require copying when consumer can reuse the passed value. It allows caller to decide whether we he needs to copy object additionally or not. With non-consuming accessor one would need to copy value inside accessor.

On the other hand as we anyway has different formats for Python and Rust strings and for other more complex objects then even consuming accessor will need to copy values to convert it into another format.

Reply:
> The reason I would prefer consuming accessor is that if into function...

That reasoning makes sense in the general case.  Although we don't have a specific internal object we're accessing aside from primitives that fit in a single CPU word and stings*.

Perhaps we could make a converting API if we find ourselves wrapping a structure that is larger and can be unwrapped without any transformation. 

> On the other hand as we anyway has different formats for Python and Rust strings and for other more complex objects then even consuming accessor will need to copy values to convert it into another format.

Exactly.  Strings would be a place where a consuming accessor could be more efficient except for the fact that there is a potential allocator mismatch and the fact that Rust strings aren't NULL-terminated.  So unfortunately I think a single-copy accessor for wrapped Rust Strings (e.g. `write into buffer`) is probably the best we can do.

Reply:
fixed in https://github.com/trueagi-io/hyperon-experimental/pull/544/commits/3a49aff93e765c634a1aeb5e9c79cce3ea5148d0